docs-md/developer-guide/ksqldb-reference/create-stream.md

```diff
@@ -13,16 +13,37 @@ Synopsis
 --------
 
 ```sql
-CREATE STREAM stream_name ( { column_name data_type } [, ...] )
+CREATE STREAM stream_name ( { column_name data_type (KEY) } [, ...] )
   WITH ( property_name = expression [, ...] );
 ```
 
 Description
 -----------
 
-Create a new stream with the specified columns and properties. Columns
-can be any of the [data types](../syntax-reference.md#ksqldb-data-types) supported by
-ksqlDB.
+Create a new stream with the specified columns and properties.
+
+A ksqlDB STREAM is a stream of _facts_. Each _fact_ is immutable and is unique.
+A stream can store its data in either `KEY` or `VALUE` columns.
+Both `KEY` and `VALUE` columns can be NULL. No special processing is done if two rows have the same
+key. The table below contrasts this to a [ksqlDB TABLE](./create-table):
+
+|                          |  STREAM                                                       | TABLE                                                             |
+| ------------------------ | --------------------------------------------------------------| ----------------------------------------------------------------- |
+| Key column type          | `KEY`                                                         | `PRIMARY KEY`                                                     |
+| NON NULL key constraint  | No                                                            | Yes <br> Messages in the Kafka topic with a NULL `PRIMARY KEY` are ignored |

```

Comment:
This is true for now (since we only support unwrapped primitive keys), but it won't be true in the future if we support single-element wrapped keys right?

Reply:
In a strict SQL sense this should remain true for wrapped and multi-column primary keys. Primary key columns in SQL must be NON NULL.

The main reason for this is that an SQL NULL is not comparable to another SQL NULL, i.e. `NULL = NULL` returns `false`.  So a `NULL` value in a primary key could never be matched.